src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreUtils.cs

```diff
@@ -77,13 +84,45 @@ public static class RestoreUtils
                     FileName = Path.GetFileName(filePath),
                     IsSnapshotRestore = snapshotRestore,
                     IsIncremental = IsIncremetal(extension),
+                    IsEncrypted = isEncrypted,
                     FilesToRestore = filesCount,
                     DatabaseName = folderDetails.DatabaseName,
                     NodeTag = folderDetails.NodeTag
                 });
             }
         }
 
+        private static bool CheckIfSnapshotIsEncrypted(string filePath, TransactionOperationContext context)
+        {
+            using (var zip = ZipFile.Open(filePath, ZipArchiveMode.Read, System.Text.Encoding.UTF8))
+            {
+                foreach (var zipEntries in zip.Entries.GroupBy(x => x.FullName.Substring(0, x.FullName.Length - x.Name.Length)))
```

Comment:
I think that we do not need to group here? Just find an entry which name equals to `RestoreSettings.SettingsFileName` ?